### PR TITLE
Rollout Jinja2 template improvements to DISP-S1 PGE

### PR DESCRIPTION
**Pull request by RKuttruff**
## Description
- Rolled out templating improvements done in #496 to DISP-S1 PGE

## Affected Issues
- Completes #524 
  - Part of #520

## Testing
- Manually diffed generated XML files from unit tests
- Ran Int test in forward mode on `opera-dev-pge` and reviewed XML output file

### CODE REVIEW

File: src/opera/pge/disp_s1/disp_s1_pge.py [hunk 1]
```diff
@@ -27,7 +28,10 @@
 from opera.util.h5_utils import get_disp_s1_product_metadata
 from opera.util.input_validation import validate_algorithm_parameters_config, validate_disp_inputs
 from opera.util.render_jinja2 import render_jinja2
-from opera.util.time import get_time_for_filename
+from opera.util.time import get_time_for_filename, get_catalog_metadata_datetime_str
+
+
+MEASURED_PARAMETER_PATH_SEPARATOR = '/'
```
Comment (collinss-jpl) [inline src/opera/pge/disp_s1/disp_s1_pge.py:34]: Is this constant defined in multiple places at this point? Maybe we should refactor to a single location in `h5_util.py`, since it's only needed when defining the measured parameter config for PGE's that produce HDF5 outputs.

File: src/opera/pge/disp_s1/disp_s1_pge.py [hunk 2]
```diff
@@ -589,6 +593,33 @@ def _qa_log_filename(self):
         """
         return self._ancillary_filename() + ".qa.log"
 
+    def augment_measured_parameters(self, measured_parameters):
```
Comment (collinss-jpl) [inline src/opera/pge/disp_s1/disp_s1_pge.py:596]: Please add a docstring for this method

File: src/opera/pge/disp_s1/disp_s1_pge.py [hunk 3]
```diff
@@ -589,6 +593,33 @@ def _qa_log_filename(self):
...
+        descriptions_file = self.runconfig.iso_measured_parameter_descriptions
+
+        new_measured_parameters = {}
+
+        if descriptions_file is not None:
```
Comment (collinss-jpl) [inline src/opera/pge/disp_s1/disp_s1_pge.py:601]: Wondering if it makes more sense to just test `if descriptions_file`, so if somehow we're passed an empty string we still trigger the error case.

File: src/opera/pge/disp_s1/disp_s1_pge.py [hunk 4]
```diff
@@ -589,6 +593,33 @@ def _qa_log_filename(self):
...
+                   'DISP-S1 metadata')
+            self.logger.critical(self.name, ErrorCode.ISO_METADATA_DESCRIPTIONS_CONFIG_NOT_FOUND, msg)
+
+        for parameter_var_name in descriptions:
+            key_path = parameter_var_name.split(MEASURED_PARAMETER_PATH_SEPARATOR)
+
+            mp = measured_parameters
+
+            while len(key_path) > 0:
+                mp = mp[key_path.pop(0)]
```
Comment (collinss-jpl) [inline src/opera/pge/disp_s1/disp_s1_pge.py:615]: This could use a try/except block that catches `KeyError` and logs a descriptive error containing the full `parameter_var_name`. Will be helpful to catch typos in the description file down the line.

File: src/opera/pge/disp_s1/disp_s1_pge.py [hunk 5]
```diff
@@ -609,6 +640,22 @@ def _collect_disp_s1_product_metadata(self, disp_product):
...
+                'ProductLevel': 3,
+                'ProductType': 'DISP-S1',
+                'ProductSource': 'Sentinel-1',
+                'ProcessingDateTime': get_catalog_metadata_datetime_str(self.production_datetime)
+            }
+
+            output_product_metadata['MeasuredParameters'] = self.augment_measured_parameters(output_product_metadata)
+
+            # Remove the hardcoded dict from main output product metadata dict once it's served its purpose
+            del output_product_metadata['static']
```
Comment (collinss-jpl) [inline src/opera/pge/disp_s1/disp_s1_pge.py:658]: Is this really necessary? Does leaving it in cause unintended side-effects? If so, would be good to know that in the comment here.

**Review by collinss-jpl — CHANGES_REQUESTED** [review]

File: src/opera/pge/disp_s1/disp_s1_pge.py (hunk 1 above)
Comment (RKuttruff) [reply src/opera/pge/disp_s1/disp_s1_pge.py]: Done

File: src/opera/pge/disp_s1/disp_s1_pge.py (hunk 2 above)
Comment (RKuttruff) [reply src/opera/pge/disp_s1/disp_s1_pge.py:594]: Done

File: src/opera/pge/disp_s1/disp_s1_pge.py (hunk 4 above)
Comment (RKuttruff) [reply src/opera/pge/disp_s1/disp_s1_pge.py]: Did this here and a couple of these other suggestions to CSLC as well as the code is identical there

File: src/opera/pge/disp_s1/disp_s1_pge.py (hunk 5 above)
Comment (RKuttruff) [reply src/opera/pge/disp_s1/disp_s1_pge.py]: I didn't really think it would be, I was thinking to minimize changes to that dictionary just in case there would be side-effects. I removed it for now

Comment (RKuttruff) [reply src/opera/pge/disp_s1/disp_s1_pge.py]: Ran a diff and there's no issues with this statement gone

**Review by collinss-jpl — APPROVED** [review]

File: src/opera/util/h5_utils.py [hunk 6]
```diff
@@ -32,6 +32,8 @@
     osr = MockOsr                           # pragma: no cover
 # pylint: enable=import-error,invalid-name
 
+MEASURED_PARAMETER_PATH_SEPARATOR = '/'
```
[the remark this answers is not in the document]
Comment (RKuttruff) [reply src/opera/util/h5_utils.py:35]: I was going to do that but I couldn't figure out how to word it...

How does this sound: 
```
Character used to delimit HDF5 metadata subgroup "paths" in measured parameter config YAML files
```

[the remark this answers is not in the document]
Comment (collinss-jpl) [reply src/opera/util/h5_utils.py:35]: Sounds great!
